hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieROTablePathFilter.java

```diff
@@ -173,6 +173,13 @@ public boolean accept(Path path) {
       }
 
       if (baseDir != null) {
+        // Check whether baseDir in nonHoodiePathCache
+        if (nonHoodiePathCache.contains(baseDir.toString())) {
+          if (LOG.isDebugEnabled()) {
+            LOG.debug("Accepting non-hoodie path from cache: " + path);
+          }
+          return true;
+        }
```

Comment:
Can you add a test case to cover this scenario? You can check existing tests in `TestHoodieROTablePathFilter`.

Reply:
Sure, will add it

Reply:
There already have the tests covered multi files have the same baseDir(`testPartitionPathsAsNonHoodiePaths`), So I just add the size check, is it enough?

Reply:
We should check this for hoodie paths as well i.e. see if files under `.hoodie` dir get accepted or not.

Reply:
Oh, I finally get your point, `.hoodie` dir get accepted or not is checked by test `testHoodiePaths`

```java
assertFalse(pathFilter.accept(new Path("file:///" + basePath + "/" + HoodieTableMetaClient.METAFOLDER_NAME + "/")));
assertFalse(pathFilter.accept(new Path("file:///" + basePath + "/" + HoodieTableMetaClient.METAFOLDER_NAME)));
```
Also add files under `.hoodie` check